connect/runtime/src/test/java/org/apache/kafka/connect/integration/ExactlyOnceSourceIntegrationTest.java

```diff
@@ -0,0 +1,1034 @@
+        propertyValidation = findConfigInfo(EXACTLY_ONCE_SUPPORT_CONFIG, validation);
+        assertFalse("Preflight validation for exactly-once support property should have at least one error message",
+                propertyValidation.configValue().errors().isEmpty());
+
+        // Connector will throw an exception from SourceConnector::exactlyOnceSupport
+        props.put("exactly.once.support.level", "fail");
+        validation = connect.validateConnectorConfig(MonitorableSourceConnector.class.getSimpleName(), props);
+        assertEquals("Preflight validation should have exactly one error", 1, validation.errorCount());
+        propertyValidation = findConfigInfo(EXACTLY_ONCE_SUPPORT_CONFIG, validation);
+        assertFalse("Preflight validation for exactly-once support property should have at least one error message",
+                propertyValidation.configValue().errors().isEmpty());
+
+        // Connector will return SUPPORTED from SourceConnector::exactlyOnceSupport
+        props.put("exactly.once.support.level", "supported");
+        validation = connect.validateConnectorConfig(MonitorableSourceConnector.class.getSimpleName(), props);
+        assertEquals("Preflight validation should have zero errors", 0, validation.errorCount());
+
+        // Test out the transaction boundary definition property
+        props.put(TRANSACTION_BOUNDARY_CONFIG, CONNECTOR.toString());
+
+        // Connector will return null from SourceConnector::canDefineTransactionBoundaries
+        props.put("custom.transaction.boundary.support", "null");
+        validation = connect.validateConnectorConfig(MonitorableSourceConnector.class.getSimpleName(), props);
+        assertEquals("Preflight validation should have exactly one error", 1, validation.errorCount());
+        propertyValidation = findConfigInfo(TRANSACTION_BOUNDARY_CONFIG, validation);
+        assertFalse("Preflight validation for transaction boundary property should have at least one error message",
+                propertyValidation.configValue().errors().isEmpty());
+
+        // Connector will return UNSUPPORTED from SourceConnector::canDefineTransactionBoundaries
+        props.put("custom.transaction.boundary.support", "unsupported");
+        validation = connect.validateConnectorConfig(MonitorableSourceConnector.class.getSimpleName(), props);
+        assertEquals("Preflight validation should have exactly one error", 1, validation.errorCount());
+        propertyValidation = findConfigInfo(TRANSACTION_BOUNDARY_CONFIG, validation);
+        assertFalse("Preflight validation for transaction boundary property should have at least one error message",
+                propertyValidation.configValue().errors().isEmpty());
+
+        // Connector will throw an exception from SourceConnector::canDefineTransactionBoundaries
+        props.put("custom.transaction.boundary.support", "null");
```

Comment:
Should this be `fail` instead of `null`?

Reply:
Good catch--yes, it should